src/rinterface_extra.c

```diff
@@ -99,6 +99,30 @@ void R_check_bool_scalar(SEXP value)
   }
 }
 
+igraph_integer_t R_get_int_scalar(SEXP sexp, int index)
```

Comment:
```suggestion
igraph_integer_t R_get_int_scalar(SEXP sexp, R_xlen_t index)
```

Or perhaps `idx_t` or similar, if it exists in C?